ArduCopter/takeoff.cpp

```diff
@@ -78,13 +87,29 @@ void Mode::_TakeOff::stop()
 //  take off is complete when the vertical target reaches the take off altitude.
 //  climb is cancelled if pilot_climb_rate_cm becomes negative
 //  sets take off to complete when target altitude is within 1% of the take off altitude
-void Mode::_TakeOff::do_pilot_takeoff(float& pilot_climb_rate_cm)
+void Mode::_TakeOff::do_pilot_takeoff(float pilot_climb_rate_cm)
 {
     // return pilot_climb_rate if take-off inactive
     if (!_running) {
         return;
     }
 
+    // calculate takeoff alt correction using range finder
+    float terr_offset = 0.0f;
+    if (use_terrain_alt) {
+        if (copter.rangefinder_alt_ok()) {
+            terr_offset = copter.inertial_nav.get_position_z_up_cm() - copter.rangefinder_state.alt_cm_filt.get();
```

Comment:
should you put a range check on this?

Reply:
Hi @andyp1per,

Thanks for the review. We don't use a range check because the terr_offset could be anything really.  For example if the vehicle took off from home and flew to the top of a mount, landed and then took off again.  One this 2nd takeoff the terr_offset could be quite large.